subsys/net/lib/shell/qbv_shell.c

```diff
@@ -0,0 +1,310 @@
+#endif
+
+	return 0;
+}
+
+/* qbv enable <iface_index> <value(off, on)> */
+static int cmd_qbv_enable(const struct shell *sh, size_t argc, char **argv)
+{
+	struct net_if *iface;
+	struct ethernet_req_params params;
+	int ret;
+	bool enable;
+	int idx;
+
+	idx = get_iface_idx(sh, argv[1]);
+	if (idx < 0) {
+		return -ENOEXEC;
+	}
+
+	iface = net_if_get_by_index(idx);
+	if (!iface) {
+		PR_WARNING("No such interface in index %d\n", idx);
+		return -ENOEXEC;
+	}
+
+	enable = shell_strtobool(argv[2], 0, &ret);
+	if (ret < 0) {
+		return ret;
+	}
+
+	params.qbv_param.type = ETHERNET_QBV_PARAM_TYPE_STATUS;
+	params.qbv_param.state = ETHERNET_QBV_STATE_TYPE_ADMIN;
+	params.qbv_param.enabled = enable;
+
+	ret = net_mgmt(NET_REQUEST_ETHERNET_SET_QBV_PARAM,
+		       iface,
+		       &params, sizeof(struct ethernet_req_params));
+	if (ret < 0) {
+		shell_error(sh, "failed to set %s", argv[1]);
+		return ret;
+	}
+
+	return 0;
+}
+
+/*
+ * qbv set_config <iface_index> <base_time(s)> <base_time(2*(-16)ns)>
+ * <cycle_time(s)> <cycle_time(ns)> <cycle_time_ext> <list_len>
+ */
+static int cmd_qbv_set_config(const struct shell *sh, size_t argc, char **argv)
+{
+	struct net_if *iface;
+	struct ethernet_req_params params;
+	uint32_t list_len;
+	int ret;
+	int idx;
+
+	idx = get_iface_idx(sh, argv[1]);
+	if (idx < 0) {
+		return -ENOEXEC;
+	}
+
+	iface = net_if_get_by_index(idx);
+	if (!iface) {
+		PR_WARNING("No such interface in index %d\n", idx);
+		return -ENOEXEC;
+	}
+
+	params.qbv_param.type = ETHERNET_QBV_PARAM_TYPE_TIME;
+	params.qbv_param.state = ETHERNET_QBV_STATE_TYPE_ADMIN;
+
+	params.qbv_param.base_time.second = shell_strtoull(argv[2], 10, &ret);
+	if (ret < 0) {
+		return ret;
+	}
+
+	params.qbv_param.base_time.fract_nsecond = shell_strtoull(argv[3], 10, &ret);
+	if (ret < 0) {
+		return ret;
+	}
+
+	params.qbv_param.cycle_time.second = shell_strtoull(argv[4], 10, &ret);
+	if (ret < 0) {
+		return ret;
+	}
+
+	params.qbv_param.cycle_time.nanosecond = shell_strtoul(argv[5], 10, &ret);
+	if (ret < 0) {
+		return ret;
+	}
+
+	params.qbv_param.extension_time = shell_strtoul(argv[6], 10, &ret);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = net_mgmt(NET_REQUEST_ETHERNET_SET_QBV_PARAM,
+		       iface,
+		       &params, sizeof(struct ethernet_req_params));
+
+	list_len = shell_strtol(argv[7], 10, &ret);
+	if (ret < 0) {
+		shell_print(sh, "failed to set times");
+		return ret;
+	}
+
+	params.qbv_param.type =
+		ETHERNET_QBV_PARAM_TYPE_GATE_CONTROL_LIST_LEN;
+	params.qbv_param.gate_control_list_len = list_len;
+	ret = net_mgmt(NET_REQUEST_ETHERNET_SET_QBV_PARAM,
+		       iface,
+		       &params, sizeof(struct ethernet_req_params));
+	if (ret < 0) {
+		shell_print(sh, "failed to set list length");
+		return ret;
+	}
+
+	return 0;
+}
+
+/* qbv set_config <iface_index> <row> <gate_control> <interval> */
+static int cmd_qbv_set_gc(const struct shell *sh, size_t argc, char **argv)
+{
+	struct net_if *iface;
```

Comment:
Could you change each sub-command to work like this:
```
static int cmd_qbv_xxx(const struct ...., )
{
#if defined(CONFIG_NET_QBV)

#else
	PR_INFO("Set %s to enable %s support.\n", "CONFIG_NET_QBV", "Qbv");
#endif
}
```
This way user is informed what options to set if Qbv is not enabled. And the code is compiled out if `CONFIG_NET_SHELL_QBV_SUPPORTED` is turned off (set to `n`, it is `y` by default).